gpt_index/indices/response/response_builder.py

```diff
@@ -375,41 +344,248 @@ async def _aget_response_tree_summarize(
             query_str, prev_response, root_nodes, text_qa_template
         )
 
-    def get_response(
+    def _get_tree_index_builder_and_nodes(
         self,
+        summary_template: SummaryPrompt,
         query_str: str,
+        text_chunks: Sequence[str],
+        num_children: int = 10,
+    ) -> Tuple[GPTTreeIndexBuilder, List[Node]]:
+        """Get tree index builder."""
+        # first join all the text chunks into a single text
+        all_text = "\n\n".join(text_chunks)
+
+        # then get text splitter
+        text_splitter = (
+            self._service_context.prompt_helper.get_text_splitter_given_prompt(
+                summary_template, num_children
+            )
+        )
+        text_chunks = text_splitter.split_text(all_text)
+        new_nodes = [Node(text=t) for t in text_chunks]
+
+        docstore = DocumentStore()
+        docstore.add_documents(new_nodes, allow_update=False)
+        index_builder = GPTTreeIndexBuilder(
+            num_children,
+            summary_template,
+            service_context=self._service_context,
+            docstore=docstore,
+            use_async=self._use_async,
+        )
+        return index_builder, new_nodes
+
+    def _get_tree_response_over_root_nodes(
+        self,
+        query_str: str,
+        prev_response: Optional[str],
+        root_nodes: Dict[int, Node],
+        text_qa_template: QuestionAnswerPrompt,
+    ) -> RESPONSE_TEXT_TYPE:
+        """Get response from tree builder over root text_chunks."""
+        node_list = get_sorted_node_list(root_nodes)
+        node_text = self._service_context.prompt_helper.get_text_from_nodes(
+            node_list, prompt=text_qa_template
+        )
+        # NOTE: the final response could be a string or a stream
+        response = super().get_response(
+            query_str=query_str,
+            text_chunks=[node_text],
+            prev_response=prev_response,
+        )
+        if isinstance(response, str):
+            response = response or "Empty Response"
+        return response
+
+class SimpleSummarize(BaseResponseBuilder):
```

Comment:
is this just trying to fit everything into one prompt? is this different than compact except for fact it'll error if prompt size exceeded?

Reply:
Doesn't Compact also run refine over chunks? Maybe I misinterpreted it.

Reply:
compact takes existing chunks, stuffs them to fit max prompt limit, so new chunks are basically prompt limit size

Reply:
`SimpleSummarize` is just logic moved from `GPTTreeIndexRetQuery`

Reply:
`SimpleSummarize` is logic moved from `GPTTreeIndexRetQuery`. Just going for no-op refactor here.